nodes/FuncDefParamsNode.java

```diff
@@ -0,0 +1,85 @@
+package nodes;
+
+import exceptions.SyntaxException;
+import provided.JottTree;
+import provided.Token;
+import provided.TokenType;
+
+import java.util.ArrayList;
+
+public class FuncDefParamsNode implements JottTree {
+
+    private ArrayList<JottTree> children = new ArrayList<>();
+
+    public FuncDefParamsNode(ArrayList<JottTree> childList){
+        try {
+            this.children.addAll(childList);
+        } catch (NullPointerException ignored) {
+            ;
+        }
+    }
+
+    @Override
+    public String convertToJott() {
+        return children.get(0).convertToJott() + ":" + children.get(1).convertToJott();
+    }
+
+    @Override
+    public String convertToJava(String className) {
+        return null;
+    }
+
+    @Override
+    public String convertToC() {
+        return null;
+    }
+
+    @Override
+    public String convertToPython() {
+        return null;
+    }
+
+    @Override
+    public boolean validateTree() {
+        return false;
+    }
+
+    public static ArrayList<FuncDefParamsNode> parseFuncDefParamsNode(ArrayList<Token> tokens) {
+        ArrayList<FuncDefParamsNode> defParamsNodes = new ArrayList<>();
+        ArrayList<JottTree> childNodeList = new ArrayList<>();
+
+        Token currToken = tokens.get(0);
+        if (currToken.getTokenType() != TokenType.ID_KEYWORD){
+            // No params for the function def. FuncDefNode will handle this token,
+            // which should be a right bracket
+            return defParamsNodes;
+        }
+        while (true) {
```

Comment:
Technically? The check for the comma at line 69 - 71 accomplishes the same thing, while also checking for the presence of another param (the case of a comma). If it's not a comma, then the current list of `FuncDefParamsNodes` are returned. Since the check for ']' occurs immediately afterwards in `FuncDefNodes.parseFuncDefNode()` (see line 85 - 89), I figured I'd avoid explicitly looking for it here. It is a part of the function_def grammar rule, after all